builder/builder-next/worker/worker.go

```diff
@@ -140,8 +141,9 @@ func NewWorker(opt Opt) (*Worker, error) {
 	}
 
 	return &Worker{
-		Opt:           opt,
-		SourceManager: sm,
+		Opt:                opt,
+		SourceManager:      sm,
+		supportedPlatforms: archutil.SupportedPlatforms,
```

Comment:
Curious; was there ea specific reason to move this to a property? (mostly because it's not a property that's fixated on initialisation, but a lookup at time of invocation)

Reply:
I needed a way to mock `archutil.SupportedPlatforms`, so that easiest thing that came to mind was making it a member of the `Worker` object, so that I could easily replace it in the test. 

Not sure if that answers your question.

Reply:
Ah! Yes, that makes sense. The only nit I could have is to possibly name it something that more clearly reflects that it's not a static list, but a lookup function, e.g. something like;

```suggestion
		platformsFn: archutil.SupportedPlatforms,
```


And for the definition, it could add a short blurb that it's to allow stubbing / mocking;

```go
type Worker struct {
	Opt
	SourceManager *source.Manager

	// platformsFn looks up supported platforms on the host,
	// and is overridden in unit-tests.
	platformsFn func(noCache bool) []ocispec.Platform
}
```

Reply:
Thanks for the suggestion, done.